contracts/RequestFactory.sol

```diff
@@ -205,4 +230,12 @@ contract RequestFactory is RequestFactoryInterface, CloneFactory {
         }
         return sign * int(windowStart - (windowStart % bucketSize));
     }
+
+    function emitStateChange(State state)
```

Comment:
*nitpick*

This seems like it would be better implemented as a modifier and to emit the events directly from the main emitter functions.  As it stands, it isn't clear that there is an access restriction to the emit functions until you dig down into this function.  The modifier approach should be easier to read and understand imho.